fetchai/ledger/transaction.py

```diff
@@ -114,6 +121,59 @@ def data(self):
     def data(self, data: bytes):
         self._data = bytes(data)
 
+    def compare(self, other: 'Transaction') -> bool:
```

Comment:
This would be cleaner, but the idea here is to only do a partial comparison. In this case, we're not comparing the signatures present, this should potentially be expanded to allow not comparing other fields (e.g.: the counter).

Reply:
The equivalence of Tx should be evaluated using the same principles as ledger uses to distinguish 2 transactions apart: everything what is included in the payload which is signed should be compared in `__eq__(...)`, what includes counter but excludes signatures.

It is different question, whether the transaction verifies (all signatures required are present & verify successfully).
The signatories are accessible via property and can be compared separately if necessary.

Reply:
Another thing is (but that goes way beyond the scope of this PR), that this semi "issue" would not come to existence if our `Transaction` class would be implemented differently - with clear separation of roles: If it would derived from `TransactionCanonical` base class (lets call it like for time being) which would carry & handle *exclusively* data members participating on payload which should be signed, and then derived `Transaction` class would just add signing functionality.

in that design, we could/would have `TransactionCacnonical.__eq__(...)` and  `Transaction.__eq__(...)` ...